packages/sn-document-viewer-react/src/components/shapes/annotation-wrapper.tsx

```diff
@@ -67,18 +63,52 @@ const useStyles = makeStyles<
 
 export const AnnotationWrapper: React.FC<Props> = (props) => {
   const permissions = useDocumentPermissions()
-
   const classes = useStyles({ ...props, permissions })
+  const annotationElement = useRef<HTMLDivElement>(null)
+  const viewerState = useViewerState()
+  const { updateState } = viewerState
+
+  useEffect(() => {
+    const handleGlobalMouseUp = () => {
+      if (viewerState.currentlyResizedElementId === props.shape.guid) {
+        updateState({ currentlyResizedElementId: undefined })
+        props.onResized(annotationElement.current?.getClientRects()[0])
+      }
+    }
+
+    document.addEventListener('mouseup', handleGlobalMouseUp)
+    return () => {
+      document.removeEventListener('mouseup', handleGlobalMouseUp)
+    }
+  }, [props, updateState, viewerState.currentlyResizedElementId])
+
+  useEffect(() => {
+    const mutationObserver = new MutationObserver((mutations) => {
+      mutations.forEach((mutation) => {
+        if (mutation.attributeName === 'style') {
+          updateState({ currentlyResizedElementId: (mutation.target as any).id })
+        }
+      })
+    })
+
+    annotationElement.current && mutationObserver.observe(annotationElement.current, { attributes: true })
+  }, [updateState])
 
   return (
     <div
-      id="annotation-wrapper"
+      id={props.shape.guid}
```

Comment:
id is not needed on the div. use props.shape.guid in the useEffect instead of (mutation.target as any).id

I think `return () => mutationObserver.disconnect()` is needed for cleanup

(please correct for other shapes too)